x-pack/plugins/apm/server/lib/apm_telemetry/index.ts

```diff
@@ -29,6 +28,45 @@ import { apmSchema } from './schema';
 
 const APM_TELEMETRY_TASK_NAME = 'apm-telemetry-task';
 
+export async function getDataTelemetry({
+  esClient,
```

Comment:
This should take a specific client. Eg internal client or currentuser client.

```suggestion
  client,
```